src/Common/HashTable/TwoLevelHashTable.h

```diff
@@ -345,4 +346,11 @@ class TwoLevelHashTable :
 
         return res;
     }
+
+    std::vector<size_t> buckets() const
+    {
+        std::vector<size_t> bucket_ids(NUM_BUCKETS);
+        std::iota(bucket_ids.begin(), bucket_ids.end(), 0);
+        return bucket_ids;
+    }
```

Comment:
It is make the overall aggregation template variants happy (all template class having `buckets()` function signature). We have another `Time` based 2 level hashtable for streaming window aggregation whose `buckets()` returns window start or end timestamps which are dynamic. For global aggregation, the `buckets()` for the two level hashtable is fixed. 

We can rename this to `getBucketIds()` for sure if you prefer that way. And in the near future, when we port the time window based streaming aggregation like `tumble(...)`, `hop(...)` and `session(...)`, we can use same name.

Reply:
> It is make the overall aggregation template variants happy (all template class having buckets() function signature). We have another Time based 2 level hashtable for streaming window aggregation whose buckets() returns window start or end timestamps which are dynamic. For global aggregation, the buckets() for the two level hashtable is fixed.

Okay then, this is perfectly reasonable.

> We can rename this to getBucketIds() for sure if you prefer that way. And in the near future, when we port the time window based streaming aggregation like tumble(...), hop(...) and session(...), we can use same name.

I have a slight preference for `getBucketIds`, but only because `buckets` is not like `size` or `empty` (which are present in a lot of stl containers), but similar to `getBufferSizeInBytes`. Not a big deal though.